components/base/kinematicbase/execution.go

```diff
@@ -219,16 +220,16 @@ func (ptgk *ptgBaseKinematics) trajectoryArcSteps(
 
 	finalSteps := []arcStep{}
 	timeStep := 0.
-	curDist := 0.
-	curInputs := []referenceframe.Input{
+	curDist := inputs[startDistanceAlongTrajectoryIndex].Value
+	startInputs := []referenceframe.Input{
 		inputs[ptgIndex],
 		inputs[trajectoryAlphaWithinPTG],
-		{curDist},
-		inputs[endDistanceAlongTrajectoryIndex],
+		inputs[startDistanceAlongTrajectoryIndex],
+		inputs[startDistanceAlongTrajectoryIndex],
```

Comment:
When creating a segment from the inputs `[index, alpha, start, end]`, the EndConfiguration is `[index, alpha, start, end]`, but the StartConfiguration will be `[index, alpha, start, start]`.

This is because that is the configuration along that arc when we start, and will be the CurrentInputs when those inputs have been sent to the base but movement has not yet started.